src/Components/Auth/ForgotPassword.jsx

```diff
@@ -0,0 +1,178 @@
+import "./styles.css";
+import { useState } from "react";
+import { useLocation } from "react-router-dom";
+import { Formik, Field, Form, ErrorMessage } from "formik";
+import * as Yup from "yup";
+import { FaEye, FaEyeSlash } from "react-icons/fa";
+import { useAuth } from "../../Context";
+
+export const ForgotPassword = () => {
+	const [isHidden, setIsHidden] = useState(true);
+	const [isAlsoHidden, setIsAlsoHidden] = useState(true);
+
+	const [isLoading, setIsLoading] = useState(false);
+	const [changedPass, setChangedPass] = useState(false);
+	const [passError, setPassError] = useState("");
+	const { state } = useLocation();
+	const previousPath = { from: state?.from ? state.from : "/" };
+
+	const { updatePassword } = useAuth();
+
+	const formSubmit = async ({
+		email,
+		password,
+		from,
+		setPassError,
+		setIsLoading,
+		setChangedPass,
+		updatePassword,
+	}) => {
+		setPassError("");
+		setIsLoading(true);
+
+		const response = await updatePassword({
+			email,
+			password,
```

Comment:
I think you can do the error handling here itself. The code seems a little redundent to me 😅